cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterStore.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.adapter;
+
+import co.cask.cdap.app.runtime.RunIds;
+import co.cask.cdap.config.Config;
+import co.cask.cdap.config.ConfigNotFoundException;
+import co.cask.cdap.config.ConfigStore;
+import co.cask.cdap.proto.Id;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import org.apache.twill.api.RunId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Store for Adapter related information.
+ */
+//TODO: Combine simple ConfigStore abstractions together.
+public class AdapterStore {
```

Comment:
dont really like that state is split up into multiple stores.  It makes it harder to make sure things are consistent and correct.  I think it would be better to store this information where status is stored.